languages/python-3/src/main/java/de/jplag/python3/Parser.java

```diff
@@ -32,25 +34,22 @@ public Parser() {
         super();
     }
 
-    public List<Token> parse(File directory, String[] files) {
+    public List<Token> parse(Set<File> files) throws ParsingException {
         tokens = new ArrayList<>();
-        errors = 0;
-        for (String file : files) {
-            logger.trace("Parsing file {}", file);
-            if (!parseFile(directory, file)) {
-                errors++;
-            }
+        for (File file : files) {
+            logger.trace("Parsing file {}", file.getName());
+            parseFile(file);
             tokens.add(Token.fileEnd(file));
         }
         return tokens;
     }
 
-    private boolean parseFile(File directory, String file) {
+    private void parseFile(File file) throws ParsingException {
         BufferedInputStream inputStream;
 
         CharStream input;
         try {
-            inputStream = new BufferedInputStream(new FileInputStream(new File(directory, file)));
+            inputStream = new BufferedInputStream(new FileInputStream(file));
```

Comment:
Not your fault, but this resource is never closed, could be a try-with-resource. Is a legacy issue though.